src/goTests.ts

```diff
@@ -0,0 +1,113 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------*/
+
+'use strict';
+
+import cp = require('child_process');
+import path = require('path');
+import vscode = require('vscode');
+import util = require('util');
+
+import { getBinPath } from './goPath';
+import { promptForMissingTool } from './goInstallTools';
+import { GoDocumentSymbolProvider } from './goOutline';
+
+export function generateTestCurrentPackage() {
+	let editor = testEditor();
+	let dir = path.dirname(editor.document.uri.fsPath);
+	generateTests({dir: dir});
+	vscode.window.showInformationMessage('gotests: Unit tests generated for package:' + path.basename(dir));
+}
+
+export function generateTestCurrentFile() {
+	let editor = testEditor();
+	let file = editor.document.uri.fsPath;
+	generateTests({dir: file});
+	vscode.window.showInformationMessage('gotests: Unit tests generated for file:' + path.basename(file));
+}
+
+export function generateTestCurrentFunction() {
+	let editor = testEditor();
+	let file = editor.document.uri.fsPath;
+
+	getFunctions(editor.document).then(functions => {
+		let currentFunction: vscode.SymbolInformation;
+		for (let func of functions) {
+			let selection = editor.selection;
+			if (selection && func.location.range.contains(selection.start)) {
+				currentFunction = func;
+				break;
+			}
+		};
+		if (!currentFunction) {
+			vscode.window.setStatusBarMessage('No function found at cursor.', 5000);
+			return;
+		}
+		generateTests({dir: file, func: currentFunction.name});
+		vscode.window.showInformationMessage('gotests: Unit test generated for function: ' + currentFunction.name + ' in file:' + path.basename(file));
+	}).then(null, err => {
+		console.error(err);
+	});
+}
+
+/**
+ * Input to goTests.
+ */
+interface Config {
+	/**
+	 * The working directory for `gotests`.
+	 */
+	dir: string;
+	/**
+	 * Specific function names to generate tests squeleton.
+	 */
+	func?: string;
+}
+
+function generateTests(conf: Config): Thenable<vscode.WorkspaceEdit> {
```

Comment:
I didn't know much about the vscode internals... I done a copy/past on the gotest plugin. I will check what it is better in my case.